docs/src/concepts.md

```diff
@@ -0,0 +1,123 @@
+
+Because a language server (usually) has an abstract syntax tree view of the file, it may be able to rename an identifier semantically. This package exposes that functionality through the hover popup, the context menu, and the top menu bar.
+
+Some language servers provide _global_ rename functionality as well. This package will present a modal dialog to ask you to confirm to apply the changes if they span more than one file.
+
+## Code Actions [new]
+
+Code Actions are an umbrella term for "Quick Fixes" and "Refactorings". They are actions that change the file (or more than one file) to resolve a diagnostic or apply a standard refactor technique. For instance, extracting a block of code into a separate method is usually called "Extract Method" and is a "Refactoring". Whereas "add a missing semicolon" would resolve a diagnostic that warns about a missing semicolon.
+
+Formatting is different from Code Actions, because Formatting is supposed to _not_ mutate the abstract syntax tree of the file, only move around white space. Any Code Action will mutate the abstract syntax tree.
+
+This package presents Code Actions as a bluish clickable annotation positioned to the right of the viewport. Alternatively, they can be presented as a light bulb in the Gutter Area.
+
+Sublime Text has no concept of Code Actions.
+
+## Code Lenses [new]
+
+Code Lenses are "actionable contextual information interspersed" in your source code.
+
+- Actionable: You can click on the link and something happens.
+- Contextual: The links are close to the code they are representing.
+- Interspersed: The links are located throughout your source code.
+
+This package presents Code Lenses as a greenish clickable annotation positioned to the right of the viewport. Alternatively, they can be presented as phantoms.
+
+Sublime Text has no concept of Code Lenses.
+
+## Server Commands
+
+In Sublime Text you can bind any runnable command to a key or add it to various UI elements. Commands in Sublime Text are normally written in the Python plugin host. A language server may provide a runnable command as well. These kinds of commands are wrapped in an `lsp_execute` Sublime command that you can bind to a key.
+
+## Server Settings
+
+Regular Sublime Text packages provide settings to adjust their behavior either via a `.sublime-settings` file that you can override in `Packages/User/package-name.sublime-settings`. A package may also allow customization through the `Packages/User/Preferences.sublime-settings` file and/or syntax-specific or project-specific settings.
+
+A language server may itself also expose settings that you can use to customize its behavior. For instance certain linter settings. This package allows customizing language server settings.
+
+## Server Initialization Options
+
+Initialization Options are like Server Settings, except they are static in the sense that they cannot be changed once the language server subprocess has started.
```

Comment:
The `Server Settings` could be a local link.

Maybe also add: "Those can be provided using the `initializationOptions` object within the client configuration" where "client configuration" could be a link to https://lsp.sublimetext.io/guides/client_configuration/.